wallet-extension/src/background/storage/activities.ts

```diff
@@ -0,0 +1,35 @@
+export interface Activity {
+  activityType: ActivityType;
+  txHash: string;
+  success: boolean;
+  amount: string;
+  token: string;
+}
+
+export enum ActivityType {
+  SEND = "send",
+  TOPUP = "top-up",
+}
+
+const STORAGE_KEY_PREFIX = "activities";
+
+// Utility to get the storage key for a wallet
+const getStorageKey = (walletAddress: string): string => {
+  return `${STORAGE_KEY_PREFIX}_${walletAddress}`;
+};
+
+// Save an activity for a specific wallet
+export const saveActivity = async (walletAddress: string, activity: Activity): Promise<void> => {
+  const storageKey = getStorageKey(walletAddress);
+  const result = await chrome.storage.local.get(storageKey);
+  const existingActivities: Activity[] = result[storageKey] || [];
+  const updatedActivities = [...existingActivities, activity];
+  await chrome.storage.local.set({ [storageKey]: updatedActivities });
+};
+
+// Get all activities for a specific wallet
+export const getActivities = async (walletAddress: string): Promise<Activity[]> => {
+  const storageKey = getStorageKey(walletAddress);
+  const result = await chrome.storage.local.get(storageKey);
```

Comment:
what is the purpose of using this `chrome` wrapper? it is some safe limited browser api access?

Reply:
`chrome.storage.local` is a part of the Chrome Extensions API, designed specifically for extensions to securely store and retrieve data. Unlike `localStorage`, it is asynchronous and allows larger storage limits (up to 5MB by default, compared to 5-10MB in localStorage). **It is isolated from regular web pages**, meaning websites cannot access extension data. This makes it safer and more suitable for browser extensions where privacy and controlled data access are important.